src/services/completions.ts

```diff
@@ -3663,71 +3664,130 @@ function getCompletionData(
                 }
             }
-            else if (preferences.includeCompletionsWithInsertText) {
-                if (firstAccessibleSymbolId && seenPropertySymbols.has(firstAccessibleSymbolId)) {
-                    return;
+            else {
+                const fileName = isExternalModuleNameRelative(stripQuotes(moduleSymbol.name)) ? getSourceFileOfModule(moduleSymbol)?.fileName : undefined;
+                const { moduleSpecifier } = (importSpecifierResolver ||= codefix.createImportSpecifierResolver(sourceFile, program, host, preferences)).getModuleSpecifierForBestExportInfo(
+                    [{
+                        exportKind: ExportKind.Named,
+                        moduleFileName: fileName,
+                        isFromPackageJson: false,
+                        moduleSymbol,
+                        symbol: firstAccessibleSymbol,
+                        targetFlags: skipAlias(firstAccessibleSymbol, typeChecker).flags,
+                    }],
+                    position,
+                    isValidTypeOnlyAliasUseSite(location),
+                ) || {};
+
+                if (moduleSpecifier) {
+                    const origin: SymbolOriginInfoResolvedExport = {
+                        kind: getNullableSymbolOriginInfoKind(SymbolOriginInfoKind.SymbolMemberExport),
+                        moduleSymbol,
+                        isDefaultExport: false,
+                        symbolName: firstAccessibleSymbol.name,
+                        exportName: firstAccessibleSymbol.name,
+                        fileName,
+                        moduleSpecifier,
+                    };
+                    symbolToOriginInfoMap[index] = origin;
                 }
-                addSymbolOriginInfo(symbol);
-                addSymbolSortInfo(symbol);
-                symbols.push(symbol);
             }
         }
-        else {
+
+        function addLiteralSymbol() {
             addSymbolOriginInfo(symbol);
             addSymbolSortInfo(symbol);
             symbols.push(symbol);
         }
 
+        /** Combines a property access expression (from the completion location to the left-most name in the computed property expression)
+         * and the computed property expression itself to return an expression to access the computed property from the completion location.
+         * Assumes that the left-most symbol is accessible so there should always be a way to access the computed property symbolically. */
+        function createComputedPropertyAccess(nameSymbol: Symbol, leftMostNameSymbol: Symbol, computedPropertyNameExpression: Expression) {
+            let node: Node | undefined;
+            if (!isTransientSymbol(nameSymbol)) {
+                node = typeChecker.symbolToEntityName(nameSymbol, /*meaning*/ undefined!, contextToken, NodeBuilderFlags.UseAliasDefinedOutsideCurrentScope);
+            }
+            else {
+                // Object literals assigned as const

```

Comment:
What's this case for, exactly? I think I don't understand the comment here and how we know this from knowing that `nameSymbol` is a transient symbol.

Reply:
This scenario goes into this branch:
```ts
const x = { a: "foo" } as const;
const y = { [x.a]: 0 };
y.|
```

It's because the symbol `a` is associated with the object and not `x` so walking up the parent symbols will just go to `__object` - so `symbolToEntityName` (the other branch) won't give a full access expression to it.

That being said, I don't know if `isTransientSymbol` is the right check here - are there transient symbols that won't run into this issue? and are there non-transient symbols that will run into this issue? I can be safe an always choose the `else` branch (get rid of the whole `if`).

Reply:
I'm not sure if all transient symbols run into this problem with `symbolToEntityName`, but I think I was confused by the comment because it sorta implied that if the symbol is transient, then we have that object literal situation, which I think is not always the case.
@weswigham might know the details of `symbolToEntityName` and `symbolToExpression`.